Add new entries on property object

7studio opened the pull request:
Hi,

This is the PR to fix the issue #327.

Finally, I choose to follow Danny's comment about naming and tests problem and publish this PR.
It is not perfect but it's a good try I think :wink:

I hope we will find the problem into `__tests__/extend.test.js` together because my brain is stuck :dizzy_face:

Thank you for your help and your encouragements.

dbanksdesign commented on lib/utils/combineJSON.js:
```diff
@@ -48,6 +58,15 @@ function combineJSON(arr, deep, collision) {
 
+    // Add some side data on each property to make filtering easier
+    traverseObj(file_content, (obj) => {
+      if (obj.value && !obj.filePath) {
```
I can't comment on lines you don't change, but we will need to change the line `file_content = require(resolvedPath);`. Because we are requiring the file rather than loading it with `fs.readFile`, any mutations we make to the object stay in memory, which is why some tests are failing and showing isSource as true when it should be false. We can fix this a few ways, first we could remove the `&& !obj.filePath` so that it will always override tokens. Although this could cause weird issues in the future because we are mutating a module. Or we could change a line above from:

```
    var file_content;
    try {
      file_content = require(resolvedPath);
```

to:

```
    var file_content = {};
    try {
      deepExtend([file_content, require(resolvedPath)]);
```

Now mutating `file_content` would not result in mutating the original module. Or there might be some more elegant solution to create a deep copy of a module.

dbanksdesign commented:
@7studio just checking in on this PR, have you had a chance to take a look at it? Let me know if you need any help!

7studio commented:
@dbanksdesign I am really sorry Danny, I fixed all the problems (all tests pass) before Christmas but I didn't take the time to push them. I rebased the branch with the last version of master and I hope I haven't forgotten something otherwise I will fix it again.

I would like to congratulate you and your partner for your baby 👶 Good luck!

7studio commented:
@dbanksdesign Have I missed something about the PR not being merged since January? If I should apply some modifications, let me know :relaxed:
